File: app/src/main/java/com/weloveyolo/moniguard/api/MoniGuardApi.java
package com.weloveyolo.moniguard.api;

import lombok.Getter;
import lombok.Setter;
import okhttp3.OkHttpClient;

@Getter
public class MoniGuardApi implements IMoniGuardApi {
    private static final String BASE_URL = "https://mgapi.bitterorange.cn";

    private String accessToken;

    private final IAnalysisApi analysisApi;

    private final IResidentsApi residentsApi;

    private final IScenesApi scenesApi;

    private final OkHttpClient httpClient;

    public MoniGuardApi() {
        httpClient = new OkHttpClient();
        this.analysisApi = new AnalysisApi(this);
        this.residentsApi = new ResidentsApi(this);
        this.scenesApi = new ScenesApi(this);
    }

    @Override
    public String getBaseUrl() {
        return BASE_URL;
    }

    @Override
    public synchronized void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }
}

File: moni_guard_api/moni_guard_api.py
from moni_guard_api.analysis_api import AnalysisApi
from moni_guard_api.scenes_api import ScenesApi


class MoniGuardApi:
    def __init__(self, url: str):
        self._access_token = ''
        self._url = url
        self._analysis_api = AnalysisApi(self._url)
        self._scenes_api = ScenesApi(self._url)

    def set_access_token(self, access_token: str):
        self._access_token = access_token
        self._analysis_api.set_access_token(access_token)

    def get_analysis_api(self) -> AnalysisApi:
        return self._analysis_api

    def get_scenes_api(self) -> ScenesApi:
        return self._scenes_api
